src/ucp/rndv/rndv.c

```diff
@@ -1414,7 +1413,7 @@ UCS_PROFILE_FUNC_VOID(ucp_rndv_receive, (worker, rreq, rndv_rts_hdr, rkey_buf),
             }
         }
 
-        if (!is_get_zcopy_failed) {
+        if (!is_get_zcopy_failed || UCP_MEM_IS_GPU(rndv_rts_hdr->mem_type)) {
```

Comment:
why do always need to register recv buffer in case get_zcopy failed? pls add/update the below comment